datasource/opennms-direct/src/main/resources/inventory.groovy

```diff
@@ -0,0 +1,180 @@
+ *
+ * Copyright (C) 2019 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2019 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.oce.datasource.opennms;
+
+import org.opennms.oce.datasource.api.InventoryObject
+import org.opennms.oce.datasource.common.ImmutableAlarm
+import org.opennms.oce.datasource.common.ImmutableInventoryObject
+import org.opennms.oce.datasource.common.inventory.ManagedObjectType;
+
+import org.opennms.integration.api.v1.model.Alarm
+import org.opennms.integration.api.v1.model.Node
+import org.opennms.integration.api.v1.model.SnmpInterface
+
+import org.opennms.oce.datasource.common.inventory.TypeToInventory;
+
+import com.google.common.base.Strings;
+
+import groovy.util.logging.Slf4j
+
+
+@Slf4j
+class InventoryFactory {
+    static List<InventoryObject> nodeToInventory(Node node) {
+        List<InventoryObject> inventoryObjects = new ArrayList<>();
+
+        InventoryObject inventoryObject = ImmutableInventoryObject.newBuilder()
+                .setType(ManagedObjectType.Node.getName())
+                .setId(toNodeCriteria(node))
+                .setFriendlyName(node.getLabel())
+                .build();
+        inventoryObjects.add(inventoryObject);
+
+        // Attach the SNMP interfaces directly to the node
+        node.getSnmpInterfaces().stream()
+                .map{snmpInterface -> toInventoryObject(snmpInterface, inventoryObject)}
+                .forEach{io -> inventoryObjects.add(io)};
+
+        return inventoryObjects;
+    }
+
+    static Collection<InventoryObject> alarmToInventory(Alarm alarm) {
+        // Only derive inventory if the alarm has an MO type and instance
+        if (Strings.isNullOrEmpty(alarm.getManagedObjectType()) ||
+        Strings.isNullOrEmpty(alarm.getManagedObjectInstance())) {
+            return Collections.emptyList();
+        }
+
+        log.info("alarmToInventory - type: {} with id: {}.", alarm.getManagedObjectType(), alarm.getManagedObjectInstance());
```

Comment:
Similarly here, can we use `log.debug` or `log.trace` instead of `log.info`?

Reply:
done